.github/workflows/deploy-prod.yaml

```diff
@@ -17,119 +17,117 @@ jobs:
+    - name: find corresponding staging release for tag
+      id: get_staging_release
       run: |
-        export VERSION_TAG=$GITHUB_REF_NAME
-        curl -H "Authorization: token $GH_PAT" \
-          -H 'Accept: application/json' \
-          -d "{\"event_type\": \"kubernetes-installer-release-notes\", \"client_payload\": {\"version\": \"${VERSION_TAG}\" }}" \
-          "https://api.github.com/repos/replicatedhq/replicated-docs/dispatches"
+        version_commit="$(git rev-list "${GITHUB_REF_NAME}" -n 1 | xargs git rev-parse --short)"
+        staging_release_dir="$(aws s3 ls s3://"${S3_BUCKET}/${STAGING_PREFIX}/" | grep "${version_commit}" | awk '{print $2}')"
+        if [ -z "${staging_release_dir}" ]; then
+            echo "Could not find staging release for tag ${GITHUB_REF_NAME} (${version_commit})"
+            return 1
+        fi
+        echo "staging_release=${staging_release_dir%/}" >> $GITHUB_OUTPUT
+      env:
+        S3_BUCKET: kurl-sh
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_PROD_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_PROD_SECRET_ACCESS_KEY }}
+        AWS_REGION: "us-east-1"
+        STAGING_PREFIX: "staging"
 
-  kurl-util-image:
+  verify-staging-release-testgrid-result:
     runs-on: ubuntu-20.04
     needs:
     - verify-tag
+    - get-staging-release
     steps:
-    - uses: actions/checkout@v3
-    - uses: docker/login-action@v2
-      with:
-        username: ${{ secrets.DOCKERHUB_DOCKER_USER }}
-        password: ${{ secrets.DOCKERHUB_DOCKER_PASS }}
-    - run: |
-        export VERSION_TAG=$GITHUB_REF_NAME
-        export KURL_UTIL_IMAGE=replicated/kurl-util:${VERSION_TAG}
-        make -C kurl_util build-and-push-kurl-util-image
-        docker tag replicated/kurl-util:${VERSION_TAG} replicated/kurl-util
-        docker push replicated/kurl-util
+    - name: Verify testgrid run
+      run: |
+        testgrid_run_json=$(curl -s "https://api.testgrid.kurl.sh/api/v1/runs?searchRef=${STAGING_RELEASE}")
+        if [ -z "${testgrid_run_json}" ]; then
+          echo "Did not find testgrid run which contains ${STAGING_RELEASE} in its Ref"
+          return 1
+        fi
 
-  build-matrix:
-    runs-on: ubuntu-20.04
-    needs:
-    - kurl-util-image
-    outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix }}
-    steps:
-    - uses: actions/checkout@v3
+        echo "Found testgrid run:"
+        echo ${testgrid_run_json} | jq
 
-    - id: set-matrix
-      name: Build image matrix
-      run: |
-        export VERSION_TAG=$GITHUB_REF_NAME
-        export KURL_UTIL_IMAGE=replicated/kurl-util:${VERSION_TAG}
-        export KURL_BIN_UTILS_FILE=kurl-bin-utils-${VERSION_TAG}.tar.gz
-        OUTPUT=`bin/list-all-packages-actions-matrix.sh "${{ github.event.inputs.index }}"`
-        echo "::set-output name=matrix::$OUTPUT"
+        tg_run_id=$(echo "${testgrid_run_json}" | jq -r '.runs[].id')
+        if ! echo "${testgrid_run_json}" | jq -e '.runs[].failure_count == 0'; then
```

Comment:
I don't think we should require no failures at this point as it is not practical.

how about checking that success_count + failure_count = total_runs?

I would suggest testing that there are no pending_runs or running_runs but it looks like if you skip runs then these wont be accounted for based on below:

```
➜  ~ curl -s "https://api.testgrid.kurl.sh/api/v1/runs?searchRef=86cfbe1f-20221220210641" | jq
{
  "runs": [
    {
      "id": "STAGING-release-v2022.12.12-0-86cfbe1f-20221220210641",
      "created_at": "2022-12-20T21:06:46.675981Z",
      "last_start": "2022-12-20T22:14:07.826523Z",
      "last_response": "2022-12-20T22:40:04.885114Z",
      "success_count": 41,
      "failure_count": 3,
      "total_runs": 98,
      "pending_runs": 0,
      "running_runs": 0
    }
  ],
  "total": 1
}
```

Reply:
See https://github.com/replicatedhq/kURL/pull/3867/files#diff-0c6dd97df28ca05e219e3f7f6cf203461efe649ca0f99d92ed4f57e1f1d6e6d6R40